impl/src/main/java/org/ehcache/impl/config/resilience/DefaultResilienceStrategyConfiguration.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.ehcache.impl.config.resilience;
+
+import org.ehcache.impl.internal.classes.ClassInstanceConfiguration;
+import org.ehcache.spi.loaderwriter.CacheLoaderWriter;
+import org.ehcache.spi.loaderwriter.CacheLoaderWriterProvider;
+import org.ehcache.spi.resilience.RecoveryStore;
+import org.ehcache.spi.resilience.ResilienceStrategy;
+import org.ehcache.spi.resilience.ResilienceStrategyProvider;
+import org.ehcache.spi.service.ServiceConfiguration;
+
+import java.util.Arrays;
+
+import static java.lang.System.arraycopy;
+import static java.lang.System.load;
+
+public class DefaultResilienceStrategyConfiguration extends ClassInstanceConfiguration<ResilienceStrategy<?, ?>> implements ServiceConfiguration<ResilienceStrategyProvider> {
+
+  public DefaultResilienceStrategyConfiguration(Class<? extends ResilienceStrategy> clazz, Object... arguments) {
+    super((Class<? extends ResilienceStrategy<?, ?>>) clazz, arguments);
+  }
+
+  public DefaultResilienceStrategyConfiguration(ResilienceStrategy<?, ?> instance) {
+    super(instance);
+  }
+
+  @Override
+  public Class<ResilienceStrategyProvider> getServiceType() {
+    return ResilienceStrategyProvider.class;
+  }
+
+  public DefaultResilienceStrategyConfiguration bind(RecoveryStore<?> store, CacheLoaderWriter<?, ?> loaderWriter) {
+    Object[] arguments = getArguments();
```

Comment:
This argument passing is fairly ugly but I guess it is the way it is...

Reply:
Yeah it's a bit nasty.  I need to also document the expected constructor form.